include/picongpu/simulation/control/Simulation.hpp

```diff
@@ -674,6 +675,16 @@ namespace picongpu
         uint32_t numRanksPerDevice = 1u;
 
     private:
+        //! Create and initialize field solver instance
+        void initFieldSolver()
+        {
+            myFieldSolver = new fields::Solver(*cellDescription);
+            // for substepping, hook the current addition stage
+            using IsSubsteppingFieldSolver = fields::maxwellSolver::traits::IsSubstepping<fields::Solver>;
+            pmacc::meta::invokeIf<IsSubsteppingFieldSolver::value>(
+                [this]() { myFieldSolver->setCurrentAddFunctor(currentInterpolationAndAdditionToEMF); });
+        }
```

Comment:
It causes warning as `setCurrentAddFunctor` is host-only.

`if constexpr` waiting room

Reply:
Maybe we need to add `PMACC_NO_NVCC_HDWARNING` over the operator implementation in `pmacc::meta::detail::IfElse<>`

Reply:
This part turned out not needed.